Network Observability 1.7 Release Notes (#81551)
openshift/openshift-docs: observability/network_observability/network-observability-operator-release-notes.adoc

[id="network-observability-cli-1-7_{context}"]
==== Network Observability CLI
The Network Observability CLI (`oc netobserv`), is now generally available. The following enhancements have been made since the 1.6 Technology Preview release:

* There are now eBPF enrichment filters for packet capture similar to flow capture.
* You can now use filter `tcp_flags` with both flow and packets capture.
* The auto-teardown option when max-bytes or max-time is hit 

For more information, see xref:../../observability/network_observability/netobserv_cli/netobserv-cli-install.adoc#network-observability-netoberv-cli-about_netobserv-cli-install[Network Observability CLI]

[id="network-observability-virtualization_{context}"]
==== Network Observability for OpenShift Virtualization

> You can observe networking patterns on an OpenShift Virtualization setup by identifying eBPF-enriched network flows coming from VMs that are connected to secondary networks, such as through Open Virtual Network (OVN)-Kubernetes. For more information, see 

Author: Yeah I need to actually add the link bc this got merged yesterday afternoon.

// link to ../../observability/network_observability/network-observability-secondary-networks.adoc#network-observability-virtualization-config_secondary_networks[Configuring virtual machine (VM) secondary network interfaces for Network Observability]
// 

[id="network-observability-network-policy-1-7_{context}"]
==== Network policy deploys in the FlowCollector custom resource (CR)
With this release, you can configure the `FlowCollector` CR to deploy a Network policy for Network Observability. Previously, if you wanted a network policy, you had to manually create one. The option to manually create a network policy is still available. For more information, see 

// link to ../../observability/network_observability/network-observability-network-policy[Configuring an ingress network policy by using the FlowCollector custom resource]

[id="network-observability-fips-compliance-1-7_{context}"]
==== FIPS compliance
* You can install and use the Network Observability Operator in an {product-title} cluster running in FIPS mode.
+
--
include::snippets/fips-snippet.adoc[]
--

[id="network-observability-dns-enhancements-1-7_{context}]

> ==== DNS tracking enhancements
> The following enhancements are available for DNS tracking:

Reviewer: ```suggestion
==== eBPF agent enhancements
The following enhancements are available for DNS tracking and agent filters:
```

(cf @stleerh comments)

> * DNS tracking can work with custom port in top of the default DNS port `53`.

Reviewer 2: This doesn't sound right.  Suggest:  
If the DNS service maps to a different port than 53, you can specify this DNS tracking port.

> * You can now use two ports for transport protocols (TCP, UDP, SCTP) filtering rules.

Reviewer 2: This is not specific to DNS tracking; it is an eBPF flow filter enhancement.  It allows you to specify up to two ports for eBPF flow filter.  Previously, you can only specify one port.

> * Filtering on transport ports can now work with a wildcard protocol by leaving the protocol field empty.

Reviewer 2: This is not specific to DNS tracking; it is an eBPF flow filter enhancement. If you specify a port but not protocol, then it filters on TCP and UDP.

Reviewer: to answer your two last comments, we can simply rename the section:
"The following enhancements are available for DNS tracking:"
into
"eBPF agent enhancements:"
?

Reviewer 2: That will work.

Author: Ok I have changed the title of this to eBPF agent enhancements, per Joel's comment below.

> * Filtering on transport ports can now work with a wildcard protocol by leaving the protocol field empty.

Reviewer 3: ```suggestion
* You can now filter on transport ports with a wildcard protocol by leaving the protocol field empty.
```

[id="network-observability-operator-1-7-bug-fixes_{context}"]
=== Bug fixes

> * Previously, when using a Rhel9.2 real-time kernel, some of the webhooks did not work. Now, a fix is in place to check if this Rhel 9.2 rt kernel is being used, and if it is, produce a warning for the features that do not work, such as packet drop and neither Round-trip Time when using `s390x` architecture. The fix is in OpenShift 4.16+. (link:https://issues.redhat.com/browse/NETOBSERV-1808[*NETOBSERV-1808*])

Reviewer 3: ```suggestion
* Previously, when using a {op-system-base} 9.2 real-time kernel, some of the webhooks did not work. Now, a fix is in place to check whether this {op-system-base} 9.2 real-time kernel is being used. If the kernel is being used, a warning is displayed about the features that do not work, such as packet drop and neither Round-trip Time when using `s390x` architecture. The fix is in OpenShift 4.16 and later. (link:https://issues.redhat.com/browse/NETOBSERV-1808[*NETOBSERV-1808*])
```

> * Previously, in the *Manage panels* dialog in the *Overview* tab, filtering on *total*, *bar*, *donut*, or *line* does not show any result. Now the available panels are correctly filtered. (link:https://issues.redhat.com/browse/NETOBSERV-1540[*NETOBSERV-1540*])

Reviewer 3: ```suggestion
* Previously, in the *Manage panels* dialog in the *Overview* tab, filtering on *total*, *bar*, *donut*, or *line* did not show a result. Now the available panels are correctly filtered. (link:https://issues.redhat.com/browse/NETOBSERV-1540[*NETOBSERV-1540*])
```

> * Previously under high stress, the eBPF agents were susceptible to enter into a state where they generated a high number of small flows, almost not aggregated. With this fix, the aggregation process is still maintained under high stress, resulting in less flows being created. This fix improves the resource consumption not only in the eBPF agent but also in `flowlogs-pipeline` and Loki. (link:https://issues.redhat.com/browse/NETOBSERV-1564[*NETOBSERV-1564*])

Reviewer 3: ```suggestion
* Previously, under high stress, the eBPF agents were susceptible to enter into a state where they generated a high number of small flows, almost not aggregated. With this fix, the aggregation process is still maintained under high stress, resulting in less flows being created. This fix improves the resource consumption not only in the eBPF agent but also in `flowlogs-pipeline` and Loki. (link:https://issues.redhat.com/browse/NETOBSERV-1564[*NETOBSERV-1564*])
```

> * Previously, when enabling the metric `workload_flows_total` instead of `namespace_flows_total`, the health dashboard stopped showing `By namespace` flow charts. With this fix, the health dashboard now shows these charts by querying the `workload_flows_total` metric. (link:https://issues.redhat.com/browse/NETOBSERV-1746[*NETOBSERV-1746*])

Reviewer 3: ```suggestion
* Previously, when the `workload_flows_total` metric was enabled instead of the `namespace_flows_total` metric, the health dashboard stopped showing `By namespace` flow charts. With this fix, the health dashboard now shows the flow charts when the `workload_flows_total` is enabled. (link:https://issues.redhat.com/browse/NETOBSERV-1746[*NETOBSERV-1746*])
```

> * Previously, when using the `FlowMetrics` API to generate a custom metric and later modifying its labels,such as adding a new label, the metric stopped populating, and an error was shown in the `flowlogs-pipeline` logs. With this fix, it is now possible to modify the labels, and the error is no longer raised in the `flowlogs-pipeline` logs. (link:https://issues.redhat.com/browse/NETOBSERV-1748[*NETOBSERV-1748*])

Reviewer 3: ```suggestion
* Previously, when you used the `FlowMetrics` API to generate a custom metric and later modified its labels, such as by adding a new label, the metric stopped populating and an error was shown in the `flowlogs-pipeline` logs. With this fix, you can modify the labels, and the error is no longer raised in the `flowlogs-pipeline` logs. (link:https://issues.redhat.com/browse/NETOBSERV-1748[*NETOBSERV-1748*])
```

> * Previously, there was an inconsistency with the default Loki `WriteBatchSize` configuration: it was set to 100KB in the `FlowCollector` CRD default, and 10MB in the OLM sample or default configuration. Both are now aligned to 10MB, which generally provides better performances and less resource footprint. (link:https://issues.redhat.com/browse/NETOBSERV-1766[*NETOBSERV-1766*])

Reviewer 3: ```suggestion
* Previously, the default Loki `WriteBatchSize` configuration was set to 100 KB in the `FlowCollector` CRD default and 10 MB in the OLM sample or default configuration. Both are now aligned to 10 MB, which generally provides better performances and less resource footprint. (link:https://issues.redhat.com/browse/NETOBSERV-1766[*NETOBSERV-1766*])
```

> * Previously, eBPF flow filter on ports without specifying a protocol were ignored. With this fix, you can set eBPF flow filters independently on ports and or protocols. (link:https://issues.redhat.com/browse/NETOBSERV-1779[*NETOBSERV-1779*])

Reviewer 3: ```suggestion
* Previously, the eBPF flow filter on ports was ignored if you did not specify a protocol. With this fix, you can set eBPF flow filters independently on ports and or protocols. (link:https://issues.redhat.com/browse/NETOBSERV-1779[*NETOBSERV-1779*])
```

> * Previously, traffic from Pods to Services was hidden from the *Topology view*; only the return traffic from Services to Pods was visible, allowing to display Services. With this fix, that traffic is correctly displayed. (link:https://issues.redhat.com/browse/NETOBSERV-1788[*NETOBSERV-1788*])

Reviewer 3: ```suggestion
* Previously, traffic from Pods to Services was hidden from the *Topology view*. Only the return traffic from Services to Pods was visible. With this fix, that traffic is correctly displayed. (link:https://issues.redhat.com/browse/NETOBSERV-1788[*NETOBSERV-1788*])
```

> * Previously, for non cluster administrator users that had access to Network Observability, an error was shown in the console plugin when they tried to filter for something that triggered auto-completion, such as a namespace. With this fix, no error is displayed, and the auto-completion returns the expected results. (link:https://issues.redhat.com/browse/NETOBSERV-1798[*NETOBSERV-1798*])

Reviewer 3: ```suggestion
* Previously, non-cluster administrator users that had access to Network Observability saw an error in the console plugin when they tried to filter for something that triggered auto-completion, such as a namespace. With this fix, no error is displayed, and the auto-completion returns the expected results. (link:https://issues.redhat.com/browse/NETOBSERV-1798[*NETOBSERV-1798*])
```

> * When the secondary interface support was added, there was a need to iterate multiple times to register the per network namespace with the netlink to learn about interface notifications. At the same time, unsuccessful handlers caused a leaking file descriptor because with TCX hook, unlike TC, handlers needed to be explicitly removed when the interface went down. Furthermore, when the network namespace was deleted, there was no go close channel event to terminate the netlinl goroutine socket, which caused go threads to leak. Now, there is no longer leaking file descriptors or go threads when creating and deleting pods. (link:https://issues.redhat.com/browse/NETOBSERV-1805[*NETOBSERV-1805*])

Reviewer 3: ```suggestion
* When the secondary interface support was added, you had to iterate multiple times to register the per network namespace with the netlink to learn about interface notifications. At the same time, unsuccessful handlers caused a leaking file descriptor because with TCX hook, unlike TC, handlers needed to be explicitly removed when the interface went down. Furthermore, when the network namespace was deleted, there was no go close channel event to terminate the netlinl goroutine socket, which caused go threads to leak. Now, there are no longer leaking file descriptors or go threads when you create or delete pods. (link:https://issues.redhat.com/browse/NETOBSERV-1805[*NETOBSERV-1805*])
```
I'm not sure whether some of the terminology here (`go close channel`, `netlinl goroutine socket`) is correctly spelled or capitalized.

Author: netlink is the interfaces events monitoring package used in the netobserv agent, and its a lowercase. 
netinl should be netlink. 
i think goroutine can be lowercase, afaict in searching google, but Go can be capitalized, since its a language.

* Previously, the ICMP type and value were displaying 'n/a' in the *Traffic flows* table even when related data was available in the flow JSON. With this fix, ICMP columns display related values as expected in the flow table. (link:https://issues.redhat.com/browse/NETOBSERV-1806[*NETOBSERV-1806*])
* Previously in the console plugin, it wasn't always possible to filter for unset fields, such as unset DNS latency. With this fix, filtering on unset fields is now possible. (link:https://issues.redhat.com/browse/NETOBSERV-1816[*NETOBSERV-1816*])

> * Previously, when clearing filters in the OpenShift web console plugin, sometimes the filters reappeared after navigating to another page and returning to the page with filters. With this fix, filters don't unexpectedly reappear after they were cleared. (link:https://issues.redhat.com/browse/NETOBSERV-1733[*NETOBSERV-1733*])

Reviewer 3: ```suggestion
* Previously, when you cleared filters in the OpenShift web console plugin, sometimes the filters reappeared after you navigated to another page and returned to the page with filters. With this fix, filters do not unexpectedly reappear after they are cleared. (link:https://issues.redhat.com/browse/NETOBSERV-1733[*NETOBSERV-1733*])
```

[id="network-observability-operator-1-7-known-issues_{context}"]
=== Known issues

> * When using the must-gather tool with Network Observability, logs aren't collected when the cluster has FIPS enabled. (link:https://issues.redhat.com/browse/NETOBSERV-1830[*NETOBSERV-1830*])

Reviewer 3: ```suggestion
* When you use the must-gather tool with Network Observability, logs are not collected when the cluster has FIPS enabled. (link:https://issues.redhat.com/browse/NETOBSERV-1830[*NETOBSERV-1830*])
```

* When the `spec.networkPolicy` is enabled in the `FlowCollector`, which installs a network policy on the `netobserv` namespace, it is impossible to use the `FlowMetrics` API. The network policy blocks calls to the validation webhook. As a workaround, use the following network policy:
+
[source,yaml]
====
kind: NetworkPolicy

> apiVersion: networking.k8s.io/v1

Author: n/a since this is part of the yaml

== Network Observability Operator 1.7.0
The following advisory is available for the Network Observability Operator 1.7.0:

* link:https://access.redhat.com/errata/RHSA-2024:8014[Network Observability Operator 1.7.0]

[id="network-observability-operator-1.7.0-features-enhancements_{context}"]
=== New features and enhancements

[id="network-observability-operator-otel-1-7_{context}"]
==== OpenTelemetry support
You can now export enriched network flows to a compatible OpenTelemetry endpoint, such as the Red Hat build of OpenTelemetry. For more information see xref:../../observability/network_observability/configuring-operator.adoc#network-observability-enriched-flows_network_observability[Export enriched network flow data].

[id="network-observability-operator-developer-perspective-1-7_{context}"]
==== Network Observability Developer perspective
You can now use Network Observability in the *Developer* perspective. For more information, see xref:../../observability/network_observability/network-observability-overview.adoc#no-console-integration[{product-title} console integration]

[id="network-observability-operator-tcp-flags-filtering-1-7_{context}"]
==== TCP flags filtering

> You can now use the `tcpFlags` filter to control the volume of packets cached in the eBPF flow table. For more information, see xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-flowcollector-flowfilter-parameters_nw-observe-network-traffic[Flow filter configuration parameters] and xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-ebpf-flow-rule-filter_nw-observe-network-traffic[eBPF flow rule filter].

Reviewer: I would rather say "to limit the volume of packets processed by the eBPF program", because there's more than caching, and I guess readers don't have to know about the internal details

```suggestion
You can now use the `tcpFlags` filter to control the volume of packets processed by the eBPF program. For more information, see xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-flowcollector-flowfilter-parameters_nw-observe-network-traffic[Flow filter configuration parameters] and xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-ebpf-flow-rule-filter_nw-observe-network-traffic[eBPF flow rule filter].
```
